packages/sdk/src/lib/dataProtectorSharing/consumeProtectedData.ts

```diff
@@ -0,0 +1,182 @@
+  const protectedData = await checkAndGetProtectedData({
+    graphQLClient,
+    protectedDataAddress: vProtectedDataAddress,
+    userAddress,
+  });
+
+  try {
+    const contentPath = '';
+    // get the app set to consume the protectedData
+    const sharingContract = await getSharingContract(
+      iexec,
+      sharingContractAddress
+    );
+    const appAddress = await sharingContract.appForProtectedData(
+      protectedData.collection.id,
+      protectedData.id
+    );
+
+    const pocoAppRegistryContract = await getPocoAppRegistryContract(iexec);
+    const appTokenId = ethers.getBigInt(appAddress).toString();
+    const appOwner = (
+      await pocoAppRegistryContract.ownerOf(appTokenId)
+    ).toLowerCase();
+    if (appOwner === DEFAULT_SHARING_CONTRACT_ADDRESS) {
+      throw new Error(
+        'The App is not owner by the protectedDataSharing Contract'
+      );
+    }
+
+    const workerpoolOrderbook = await iexec.orderbook.fetchWorkerpoolOrderbook({
+      workerpool: WORKERPOOL_ADDRESS,
+      app: appAddress,
+      dataset: vProtectedDataAddress,
+      minTag: SCONE_TAG,
+      maxTag: SCONE_TAG,
+    });
+    const workerpoolOrder = workerpoolOrderbook.orders[0]?.order;
+    if (workerpoolOrder.workerpoolprice > 0) {
+      throw new Error('No workerpool order free');
+    }
+
+    const tx = await sharingContract.consumeProtectedData(
+      protectedData.collection.id,
+      protectedData.id,
+      workerpoolOrder,
+      contentPath,
+      {
+        // TODO: See how we can remove this
+        gasLimit: 900_000,
+      }
+    );
+    await tx.wait();
+    return {
+      success: true,
+      txHash: tx.hash,
+    };
+  } catch (e) {
+    throw new WorkflowError(
+      'Sharing smart contract: Failed to subscribe to collection',
+      e
+    );
+  }
+};
+
+async function checkAndGetProtectedData({
+  graphQLClient,
+  protectedDataAddress,
+  userAddress,
+}: {
+  graphQLClient: GraphQLClient;
+  protectedDataAddress: Address;
+  userAddress: Address;
+}) {
+  const { protectedData } = await getProtectedDataById({
+    graphQLClient,
+    protectedDataAddress,
+  });
+
+  if (!protectedData) {
+    throw new ErrorWithData(
+      'This protected data does not exist in the subgraph.',
+      { protectedDataAddress }
+    );
+  }
+
+  if (protectedData.owner.id !== DEFAULT_SHARING_CONTRACT_ADDRESS) {
+    throw new ErrorWithData(
+      'This protected data is not owned by the sharing contract, hence a sharing-related method cannot be called.',
+      {
+        protectedDataAddress,
+        currentOwnerAddress: protectedData.owner.id,
+      }
+    );
+  }
+
+  if (protectedData.collection?.owner?.id !== userAddress) {
+    throw new ErrorWithData(
+      'This protected data is not part of a collection owned by the user.',
+      {
+        protectedDataAddress,
+        currentCollectionOwnerAddress: protectedData.collection?.owner?.id,
+      }
+    );
+  }
```

Comment:
Test irrelevant here?

It's fine if the protected data is in another collection, you could still consume it if you have rented it for example.

It's also fine if the protected data is in a collection of yours, you might want to consume your own protected data, anytime and for free.

Reply:
yeah you are right my mistake